File: frontend/oei-web/src/app/domain/model/institution/institution-opportunity.ts
// Les 6 types d'opportunité (doc 03 §"Opportunités").
export type InstitutionOpportunityType = 'JOB' | 'INTERNSHIP' | 'MENTORING' | 'PRO_BONO' | 'WORKING_GROUP' | 'CALL_FOR_EXPERTS';

export type InstitutionOpportunityStatus = 'DRAFT' | 'PUBLISHED' | 'CLOSED' | 'REPORTED';

export interface InstitutionOpportunity {
  readonly id: string;
  readonly institutionId: string;
  readonly type: InstitutionOpportunityType;
  readonly title: string;
  readonly description: string;
  readonly status: InstitutionOpportunityStatus;
  readonly expiresAt: string | null;
  readonly publishedAt: string | null;
}

export interface InstitutionOpportunityCreation {
  readonly type: InstitutionOpportunityType;
  readonly title: string;
  readonly description: string;
  readonly expiresAt: string | null;
}

export function createInstitutionOpportunity(fields: InstitutionOpportunity): InstitutionOpportunity {
  return Object.freeze({ ...fields });
}

File: backend/domain/shared/src/main/java/global/oei/domain/shared/institution/InstitutionOpportunity.java
package global.oei.domain.shared.institution;

import java.time.Instant;
import java.util.Objects;

/**
 * An opportunity (job/internship/mentoring/pro-bono/working-group/call-for-experts) published
 * by an institution. Published immediately on creation in this iteration — the contract
 * defines no separate draft-submission workflow for opportunities, only {@link #close}.
 */
public record InstitutionOpportunity(
        String id,
        InstitutionId institutionId,
        InstitutionOpportunityType type,
        String title,
        String description,
        Instant expiresAt,
        InstitutionOpportunityStatus status,
        Instant publishedAt) {

    public InstitutionOpportunity {
        Objects.requireNonNull(id, "id must not be null");
        Objects.requireNonNull(institutionId, "institutionId must not be null");
        Objects.requireNonNull(type, "type must not be null");
        Objects.requireNonNull(title, "title must not be null");
        Objects.requireNonNull(description, "description must not be null");
        Objects.requireNonNull(status, "status must not be null");
    }

    public InstitutionOpportunity withContent(
            final InstitutionOpportunityType type, final String title, final String description, final Instant expiresAt) {
        return new InstitutionOpportunity(id, institutionId, type, title, description, expiresAt, status, publishedAt);
    }

    /**
     * @return a new instance moved to {@link InstitutionOpportunityStatus#CLOSED}
     */
    public InstitutionOpportunity close() {
        return new InstitutionOpportunity(id, institutionId, type, title, description, expiresAt, InstitutionOpportunityStatus.CLOSED, publishedAt);
    }
}
